matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/task/TaskExecutor.kt

```diff
@@ -31,39 +32,56 @@ import timber.log.Timber
 import javax.inject.Inject
 import kotlin.coroutines.EmptyCoroutineContext
 
-internal class TaskExecutor @Inject constructor(private val coroutineDispatchers: MatrixCoroutineDispatchers) {
+@MatrixScope
+internal class TaskExecutor @Inject constructor(private val coroutineDispatchers: MatrixCoroutineDispatchers,
+                                                private val networkConnectivityChecker: NetworkConnectivityChecker) {
+
+    private val cancelableBag = CancelableBag()
```

Comment:
This can be replaced with `private val executorScope = CoroutineScope(SupervisorJob())`, then this can be used instead of `GlobalScope`.

Reply:
Thanks for all the advices on coroutines, we are not aware of all the available methods

Reply:
No problem.